items:
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception
  id: _key_vault_error_exception
  parent: com.microsoft.azure.keyvault.models
  children:
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.getBody()
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.getResponse()
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException()
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final String)
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final String,final Throwable)
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final Throwable)
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.setBody(KeyVaultError)
  - com.microsoft.azure.keyvault.models._key_vault_error_exception.setResponse(Response)
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: KeyVaultErrorException
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException
  type: Class
  source:
    remote: &o0
      path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
      branch: master
      repo: https://github.com/Azure/azure-sdk-for-java.git
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 18
  package: com.microsoft.azure.keyvault.models
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.keyvault.models._key_vault_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: &o1
    content: public class KeyVaultErrorException extends RestException
  inheritance:
  - java.lang.Object
  - Exception
  - com.microsoft.rest._rest_exception
  inheritedMembers:
  - com.microsoft.rest._rest_exception.RestException()
  - com.microsoft.rest._rest_exception.RestException(String)
  - com.microsoft.rest._rest_exception.RestException(Throwable)
  - com.microsoft.rest._rest_exception.RestException(String,Throwable)
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.getBody()
  id: getBody()
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: getBody()
  fullName: KeyVaultError com.microsoft.azure.keyvault.models.KeyVaultErrorException.getBody()
  type: Method
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 69
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Gets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public KeyVaultError getBody()
    return:
      type: com.microsoft.azure.keyvault.models._key_vault_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.getResponse()
  id: getResponse()
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: getResponse()
  fullName: Response com.microsoft.azure.keyvault.models.KeyVaultErrorException.getResponse()
  type: Method
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 61
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Gets information about the associated HTTP response.</p>

    <p></p>
  syntax:
    content: public Response getResponse()
    return:
      type: e9d34ed6
      description: <p>the HTTP response </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException()
  id: KeyVaultErrorException()
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: KeyVaultErrorException()
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException.KeyVaultErrorException()
  type: Constructor
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 30
  package: com.microsoft.azure.keyvault.models
  summary: <p>Initializes a new instance of the <xref href="com.microsoft.azure.keyvault.models._key_vault_error_exception" data-throw-if-not-resolved="false"></xref> class. </p>
  syntax:
    content: public KeyVaultErrorException()
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final String)
  id: KeyVaultErrorException(final String)
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: KeyVaultErrorException(final String message)
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException.KeyVaultErrorException(final String message)
  type: Constructor
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 36
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.keyvault.models._key_vault_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public KeyVaultErrorException(final String message)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>The exception message. </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final String,final Throwable)
  id: KeyVaultErrorException(final String,final Throwable)
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: KeyVaultErrorException(final String message, final Throwable cause)
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException.KeyVaultErrorException(final String message, final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 45
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.keyvault.models._key_vault_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public KeyVaultErrorException(final String message, final Throwable cause)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>the exception message </p>
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.KeyVaultErrorException(final Throwable)
  id: KeyVaultErrorException(final Throwable)
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: KeyVaultErrorException(final Throwable cause)
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException.KeyVaultErrorException(final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 53
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.keyvault.models._key_vault_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public KeyVaultErrorException(final Throwable cause)
    parameters:
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.setBody(KeyVaultError)
  id: setBody(KeyVaultError)
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: setBody(KeyVaultError body)
  fullName: void com.microsoft.azure.keyvault.models.KeyVaultErrorException.setBody(KeyVaultError body)
  type: Method
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 85
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Sets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public void setBody(KeyVaultError body)
    parameters:
    - id: body
      type: com.microsoft.azure.keyvault.models._key_vault_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception.setResponse(Response)
  id: setResponse(Response)
  parent: com.microsoft.azure.keyvault.models._key_vault_error_exception
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  langs:
  - java
  name: setResponse(Response response)
  fullName: void com.microsoft.azure.keyvault.models.KeyVaultErrorException.setResponse(Response response)
  type: Method
  source:
    remote: *o0
    path: azure-keyvault/src/main/java/com/microsoft/azure/keyvault/models/KeyVaultErrorException.java
    startLine: 77
  package: com.microsoft.azure.keyvault.models
  summary: >-
    <p>Sets the HTTP response.</p>

    <p></p>
  syntax:
    content: public void setResponse(Response response)
    parameters:
    - id: response
      type: e9d34ed6
      description: <p>the HTTP response </p>
references:
- uid: 3d93591e
  spec.java:
  - name: final String
    fullName: final String
- uid: fa00b19f
  spec.java:
  - name: final Throwable
    fullName: final Throwable
- uid: e9d34ed6
  spec.java:
  - name: Response
    fullName: Response
- uid: com.microsoft.rest._rest_exception.RestException()
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException()
  fullName: com.microsoft.rest.RestException.RestException()
  type: Constructor
  summary: <p>Initializes a new instance of the AutoRestException class. </p>
  syntax:
    content: public RestException()
- uid: com.microsoft.rest._rest_exception.RestException(String)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message)
  fullName: com.microsoft.rest.RestException.RestException(String message)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message)
    parameters:
    - id: message
      type: "26831127"
      description: <p>The exception message. </p>
- uid: com.microsoft.rest._rest_exception.RestException(Throwable)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(Throwable cause)
    parameters:
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception.RestException(String,Throwable)
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message, Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(String message, Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message, Throwable cause)
    parameters:
    - id: message
      type: "26831127"
      description: <p>the exception message </p>
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception
  parent: com.microsoft.rest
  href: com.microsoft.rest._rest_exception.yml
  name: RestException
  fullName: com.microsoft.rest.RestException
  type: Class
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class RestException
- uid: com.microsoft.azure.keyvault.models._key_vault_error
  parent: com.microsoft.azure.keyvault.models
  href: com.microsoft.azure.keyvault.models._key_vault_error.yml
  name: KeyVaultError
  fullName: com.microsoft.azure.keyvault.models.KeyVaultError
  type: Class
  summary: <p>the key vault error exception. </p>
  syntax:
    content: public class KeyVaultError
- uid: com.microsoft.azure.keyvault.models._key_vault_error_exception
  parent: com.microsoft.azure.keyvault.models
  href: com.microsoft.azure.keyvault.models._key_vault_error_exception.yml
  name: KeyVaultErrorException
  fullName: com.microsoft.azure.keyvault.models.KeyVaultErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.keyvault.models._key_vault_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: *o1
